lib/qa/authorities/linked_data/config.rb

```diff
@@ -0,0 +1,77 @@
+require 'qa/authorities/linked_data/config/term_config'.freeze
+require 'qa/authorities/linked_data/config/search_config'.freeze
+
+module Qa::Authorities
+  module LinkedData
+    class Config
+      attr_reader :authority_name
+      attr_reader :authority_config
+
+      # Initialize to hold the configuration for the specifed authority.  Configurations are defined in config/authorities/linked_data.  See README for more information.
+      # @param [String] the name of the configuration file for the authority
+      # @return [Qa::Authorities::LinkedData::Config] instance of this class
+      def initialize(auth_name)
+        @authority_name = auth_name
+        auth_config
+      end
+
+      include Qa::Authorities::LinkedData::TermConfig
+      include Qa::Authorities::LinkedData::SearchConfig
+
+      # Return the full configuration for an authority
+      # @return [String] the authority configuration
+      def auth_config
+        @authority_config ||= LINKED_DATA_AUTHORITIES_CONFIG[@authority_name]
+        raise Qa::InvalidLinkedDataAuthority, "Unable to initialize linked data authority #{@authority_name}" if @authority_config.nil?
+        @authority_config
+      end
+
+      private
+
+        # Removes characters from the query string that are not tolerated by the API
+        #   See oclc sample code at
+        #   http://experimental.worldcat.org/fast/assignfast/js/assignFASTComplete.js
+        def clean_query_string(q)
+          URI.escape(q.gsub(/-|\(|\)|:/, ""))
```

Comment:
Should the gsub be done without the users knowledge? Would it perhaps be better to raise an error if it matched one of those restricted characters so the user can fix their query to this API? Or maybe I am misunderstanding the method and this isn't related to the users query string.